### PR TITLE
fix(ext/kv): same `expireIn` should generate same `expireAt`

### PR DESCRIPTION
**Pull request by losfair**

### CODE REVIEW

**Review by lucacasonato — COMMENTED** [review]
We should compute this on `commit` not on AO creation I think.

File: ext/kv/01_db.ts [hunk 1]
```diff
@@ -419,11 +421,26 @@ class AtomicOperation {
...
+      const now = Date.now();
+      mutations = mutations.map(([key, type, value, expireIn]) => {
+        return [
+          key,
+          type,
+          value,
+          expireIn !== undefined ? now + expireIn : undefined,
+        ];
+      });
+    }
```
Comment (lucacasonato) [inline ext/kv/01_db.ts:437]: We do two loops over the array here (findIndex and map) - can you reduce this to one?

Comment (losfair) [reply ext/kv/01_db.ts:437]: If we do one loop (construct the patched `mutations` array directly), there will be unnecessary allocation if there is no keys with expiration in this atomic op. Unless we overwrite `this.#mutations` which I don't want to do (would be confusing if the user accidentally committed twice).

Comment (lucacasonato) [reply ext/kv/01_db.ts:437]: I pushed a patch to simplify the logic

**Pull request merged**